.cirrus.yml

```diff
@@ -269,9 +302,15 @@ qa_task:
     echo 'Recording tests on video'
     ffmpeg -loglevel warning -f x11grab -video_size 1280x960 -i ${DISPLAY} -codec:v libx264 -r 12 ${CIRRUS_WORKING_DIR}/recording_${IDEA_VERSION}.mp4
   run_its_script: |
-    echo "Run ITs on ${IDEA_VERSION}"
+    echo "Run ITs on ${IDEA_VERSION} with category: ${QA_CATEGORY}"
     source .cirrus/use-gradle-wrapper.sh
-    gradle :its:check --stacktrace -i -PijVersion=${IDEA_VERSION} -PslPluginDirectory=${CIRRUS_WORKING_DIR}/staged-plugin
```

Comment:
I'm a bit confused, to be honest. It seems at least from what you changed, that not all integration tests will run anymore except for `StandaloneIdeaTests`, `CurrentFileTests`, and `HotspotVulnerabilityScalaTests` (formerly `ConnectedIdeaTests`). Is that the case?

Reply:
Hi Tobias,

Currently we have 6 test Classes:
ClionTests
CurrentFileTests
GoSonarLanguageTestsNode
HotspotVulnerabilityScalaTests
PLSQLTests
StandaloneTests

Parallelization is only applied to tests running on IC.

Clion,PLSQL and Go tests can not be run on IC.
CLion and Go requires Clion and GoLand IDEs
PLSQL requires Databases and SQL plugin which is not present on IC

We are not running any tests on Intellij Ultimate at the moment.

So, in short, we are indeed running all the tests that can be run on IC.

Reply:
We had a quick talk. Due to the bash if statement, on all axes there are not all tests executed anymore which has to be changed back.

Reply:
Yes, added another elif statement and the following else that executes all tests for the CLion and GoLand IDEs to cover this case.

Reply:
Good addition but the else should then not run `StandaloneIdeaTests`, `HotspotVulnerabilityScalaTests`, `CurrentFileTests`.
I'm only aware that you can exclude test classes via a configuration in the `build.gradle.kts` that is triggered like a Maven profile on the call.

Reply:
The Else is run for the IDEs Clion and GoLand. We have the annotation @DisabledIf("isCLionOrGoLand") on those classes to skip those tests. In fact, you can verify this on the cirrus CI you will see :(StandaloneIdeaTests > should_exclude_rule_() SKIPPED). I changed the code to use check instead of test to be consistent with previous versions.

Reply:
You could leverage the matrix here instead of defining if-else clauses. The purpose of the matrix is to create variations, and the list of excluded test suites is the variation here